### DISCUSSION TITLE
Rsync from remote to remote

### DISCUSSION THREAD

**Discussion by 404erreur** [RsyncProject/rsync#436] [Q&A]
Hello,

I would like to know if rsync can synchronise  2 remotes directorys on seperated server (servers not able to communicate)
Like an "scp -3 -r" but with the rsync perf ?

Using sshfs or something like that can't help me, I really want using rsync directly to make this synchronitation.

I all ready find some post with some triks, but I guess rsync can do it himself and properly :)

Many thanks if anyone can reply :).

Best regards
404

**Comment by michaelmess-de** [comment]
Having two remotes would be a desireable feature.

Rsync can only have one remote process (started as --server) which the client connects to.
The remote can be either the source or target (first or second path), if it is the source it gets the --sender argument. 
rsync can also copy locally (there is no remote) from local to local. In this case it forks a local server process.

To support having two remotes, one of them would have to act like an initiator from the protocol view and as a server for accepting a connection (--server --sender), the other one acts as a receiver (--server) . 
The protocol data would then have to be routed by the local process between the two remote processes and the local process would terminate when both remote processes are finished with their work.

I think, it should be possible, but maybe needs some refactoring for process startup and requires intensive testing in order not to break anything.
